cmake/CPackSetup.cmake

```diff
@@ -0,0 +1,60 @@
+# package info
+set(CPACK_PACKAGE_NAME atsdk)
+set(CPACK_PACKAGE_DESCRIPTION "Atsign's atSDK library")
+set(CPACK_PACKAGE_VERSION 0.3.0)
+set(CPACK_PACKAGE_VENDOR_NAME atsign-foundation)
+
+# cmake configuration
+set(CPACK_CMAKE_GENERATOR "Unix Makefiles")
+set(
+  CPACK_PACKAGE_INSTALL_DIRECTORY
+  "${CMAKE_CURRENT_SOURCE_DIR}/package-source"
+)
+
+# static cpack configuration
+set(CPACK_PACKAGE_FILE_NAME atsdk-static)
+set(CPACK_GENERATOR ZIP TGZ)
+set(CPACK_INSTALL_CMAKE_PROJECTS ".;atsdk;ALL;/")
+
+# source cpack configuration
+set(CPACK_SOURCE_PACKAGE_FILE_NAME atsdk-source)
+set(CPACK_SOURCE_GENERATOR TGZ ZIP)
+
+set(CPACK_SOURCE_INSTALLED_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR};/")
+
+# Yes, you need 4 * '\' to escape literal '.'
+set(
+  CPACK_SOURCE_IGNORE_FILES
+  # directories
+  "/\\\\.cache/"
+  "/\\\\.git/"
+  "/\\\\.github/"
+  "/\\\\.jj/"
+  "/\\\\.vscode/"
+  "/CMakeFiles/"
+  "/docs/"
+  "/build/"
+  "/generators/"
+  "/package-source/"
+  # root files
+  "/\\\\.clang-format"
+  "/\\\\.clang-tidy"
+  "/\\\\.clangd"
+  "/just.env"
+  "/just.template.env"
+  "/justfile"
+  "/valgrind.Dockerfile"
+  # (Potentially) nested files
+  "\\\\.DS_Store"
+  "\\\\.gitignore"
+  "compile_commands.json"
+)
+
+set(CPACK_INSTALL_SCRIPT "${CMAKE_CURRENT_BINARY_DIR}/CPackSourceDeps.cmake")
+configure_file(
+  "${CMAKE_SOURCE_DIR}/cmake/CPackSourceDeps.cmake"
+  "CPackSourceDeps.cmake"
+  @ONLY
+)
```

Comment:
Added these lines to run an additional script which copies over mbedtls and cjson src...

but it also does so for the static build, I _could_ spend time to figure out how to conditionally do this, but there is virtually no documentation for cpack variables... even some of the ones being used here aren't documented... so I'm not going to spend time on something we don't care about right now.

As for why we can't do this with the exclude list:
Unfortunately there is no way in cmake to specify inverted regular expressions, and generating a recursive globular filter is awful. I already tried, but we need to go multiple levels deep, and performance was already terrible.